aws/test/colored-money-hardhat/test/RLN.test.ts

```diff
@@ -2,11 +2,47 @@ import { expect } from 'chai';
 import { UserFactory } from 'nightfall-sdk';
 import { UserConfig } from '../helper-hardhat-config';
 
+const getRLNBalance = async (
+  user: any,
+  contractAddress: string,
+  entityId: string,
+  onlyConfirmed: boolean = false,
+) => {
+  let balance = 0;
+  let balanceTokenPendingDeposits = 0;
+  let balanceTokenNightfall = 0;
+
+  if (!onlyConfirmed) {
+    try {
+      const pendingDeposits = await user.checkPendingDeposits();
+      balanceTokenPendingDeposits =
+        pendingDeposits[contractAddress.toLowerCase()].find((t: { tokenId: string }) =>
+          t?.tokenId?.includes(entityId),
+        )?.balance || 0;
```

Comment:
The type is not correct. I would remove it, use `unknown` or declare it at the top of the file. Maybe...

```ts
interface TokenIdBalance {
  balance: number,
  token: string,
}
```

```suggestion
        pendingDeposits[contractAddress.toLowerCase()].find((t: number | TokenIdBalance ) =>
          t?.tokenId?.includes(entityId),
        )?.balance || 0;
```

It is `number` for the first element of the array I believe.

Reply:
It's not working with this approach and get an error about tokenId is not a property of number. So I leave it as it is.